### ISSUE TITLE
Milnerton & Birch Acres Issues

### ISSUE THREAD

**Issue by uncappedshady** [beyarkay/eskom-calendar#55]
Hi there, do you perhaps have an ics file for Milnerton WesternCape?

Thanks

**Comment by beyarkay** [comment]
Hey! Milnerton is in the city of cape town area 2 (If i remember correctly) so go to [the releases page](https://github.com/beyarkay/eskom-calendar/releases/tag/latest) and search for 
city-of-cape-town-area-2.ics. You can rename it in your calendar application to "Milnerton" if you want to 😁 

Feel free to re-open the issue if this doesn't solve your question.

**Comment by uncappedshady** [comment]
Thank you so much for this. I have moved away from another integration as it was not reliable. I have integrated Google Calendar into my Home Assistant and it works a treat. 
![Uploading 19FA8E53-4275-4E7A-BC8F-1A781FA6F511.png…]()

**Comment by uncappedshady** [comment]
![1596702E-919A-4837-9988-0037AB2A4358](https://user-images.githubusercontent.com/20617622/190867076-ab5f177d-5714-413d-93ee-f26847ad9858.png)

**Comment by uncappedshady** [comment]
Thank you for your assistance. I was actually looking for Birch Acres Block 14's loadshedding times and Milnerton used to be the same time slots as Birch Acres.
I saw that you actually had Birch Acres under Gauteng but it seems that your end times are 30 minutes short of ending for example: 
Your .ics calendar shows it is 02:00 - 04:00 but it is actually 02:00 - 04:30 here in Birch Acres.
I looked at another loadshedding app and it shows it correctly so not sure why Eskom is reporting it wrong where you scrape the info from. Is there a solution to "fix" this using your .ics file?

I took the liberty of taking down all the loadshedding times with their stages and save the info in a MySQL database up until december I think. I will attach a .sql dump here incase that can give you an idea of the time slots. Not sure where the other loadshedding app provider is getting his details from but all I know is that it is accurate for Gauteng, Birch Acres Block 14.

Appreciate the help. 

I cannot seem to upload the .sql file so I just attached the txt file. Hope that helps.

[loadshedding_birch_acres_b_14_txt.txt](https://github.com/beyarkay/eskom-calendar/files/9592625/loadshedding_birch_acres_b_14_txt.txt)

**Comment by beyarkay** [comment]
I'm glad this is working for you! can I ask how you heard about eskom-calendar?

To answer your questions:

> I saw that you actually had Birch Acres under Gauteng...

Is Birch Acres not under Gauteng? I can only find [this search result](https://goo.gl/maps/hULJw2zx26e4twuFA) on google maps, so I'm not sure if we're talking about the same place?

> Your .ics calendar shows it is 02:00 - 04:00 but it is actually 02:00 - 04:30 here in Birch Acres

Hmm, that's weird. `ekurhuleni` is a fairly recent addition, and in chatting with another user it seemed as though loadshedding only came in two hour increments. See [this comment](https://github.com/beyarkay/eskom-calendar/issues/41#issuecomment-1245881125) for context, or [the original PDF](https://www.ekurhuleni.gov.za/for-me/load-shedding-schedules/) which lists loadshedding as being 2 hours long. Can you confirm that loadshedding is 2h30m long in `gauteng-ekurhuleni-block-14`, going against what the PDF says?


As an aside, since you're tech savy you might like this line below. It'll download the machine-friendly version of every single calendar available and then `grep` it for lines that contain the area "gauteng-ekurhuleni-block-14" and the time "T02:00:00". So from this I can confirm that you are seeing loadshedding times from 2022-09-19T02:00:00 to 2022-09-19T04:00:00 like you said.

```
$ curl -Ls https://github.com/beyarkay/eskom-calendar/releases/download/latest/machine_friendly.csv \
     | grep "gauteng-ekurhuleni-block-14" \
     | grep "T02:00:00"
gauteng-ekurhuleni-block-14,2022-09-19T02:00:00+02:00,2022-09-19T04:00:00+02:00,6,"https://twitter.com/Eskom_SA/status/1571328907005120513"
gauteng-ekurhuleni-block-14,2022-09-20T02:00:00+02:00,2022-09-20T04:00:00+02:00,6,"https://twitter.com/Eskom_SA/status/1571328907005120513"
```

**Comment by uncappedshady** [comment]
Thank you for the reply. So I used the Loadshedding notifier app before integrating loadshedding into hassio and I can say that our schedule for Gauteng, Ekhuruleni, Birch Acres Block 14 follows their schedule to a tee. Please see the screenshot of the app for tomorrows loadshedding schedule:
![Uploading DAB7351C-40D9-4C79-A067-B64104307CAE.png…]()

**Comment by uncappedshady** [comment]
I had a look at that pdf and it was last updated in Nov 2021. I also read through your linked page with the comments and all I can tell you is that Birch Acres Block 14 which I fall under follows the schedule on the Notifier App. Also to comment on your question where I heard about eskom-calander? I was scouring the web and Github for  *loadshedding or *eskom and happened on your page. Also if you correlate the data I sent you in the text file, that was manually captured from the App I am using.

**Comment by beyarkay** [comment]
> If I need to adjust an ics file to add 30 mins to each ending time, would that be difficult to do?

For you it would be tricky (since you wouldn't be able to get any updates made to the calendar), but it's pretty easy for me to make the fix, I just want to make sure I've got the data correct before I do so. I'll look through the data on the Notifier App and make sure my data matches up with theirs for Birch Acres Block 14 so that you're all sorted 👍 (should be able to do that tonight)

> I had a look at that pdf and it was last updated in Nov 2021

Yeah this is a common issue. If I don't know anyone in the area its actually really tricky to figure out what the schedules are from the internet alone.

> Birch Acres has issues with their substations and they switch suburbs on in phases so maybe what I am seeing is the delay in getting our area back up

That's interesting. But as far as I can tell you're the only one using `gauteng-ekurhuleni-block-14.ics` (check [this link](https://somsubhra.github.io/github-release-stats/?username=beyarkay&repository=eskom-calendar&page=1&per_page=5) and then search for `gauteng-ekurhuleni-block-14.ics`, you'll see there's only one download) so I'm happy to change the schedule to whatever works for you.

**Comment by uncappedshady** [comment]
Really appreciate it bud. I would suggest to leave the original.ics and duplicate it with the 30min offset and maybe call it gauteng-ekurhuleni-block-14-30-min-offset.ics
That way if they change the times to suit what others are experiencing, I can just revert back to that one? 

Truly appreciate your efforts and a big thank you to you kind sir.

**Comment by beyarkay** [comment]
Should be fixed by [5c1f22b](https://github.com/beyarkay/eskom-calendar/commit/5c1f22b0b99e3ab5694d1ba532c1256a91532058). If you don't have any other questions / comments / suggestions then I'll close this issue in a few days.

**Comment by uncappedshady** [comment]
Thank you so much for the speedy resolve. Much appreciated!
